src/firmware/app/control/trajectory_planner.c

```diff
@@ -421,19 +562,106 @@ void app_trajectory_planner_generateVelocityTrajectory(
     const float delta_y = position_elements[last_element_index].position.y -
                           position_elements[last_element_index - 1].position.y;
 
+    float x_velocity_component = 0;
+    float y_velocity_component = 0;
+
     // Get the inverse magnitude of the direction vector for normalization
     const float vector_magnitude_inverse = (1 / sqrt(pow(delta_x, 2) + pow(delta_y, 2)));
-    const float x_velocity_component =
-        vector_magnitude_inverse * delta_x *
-        position_trajectory->speed_profile[last_element_index];
-    const float y_velocity_component =
-        vector_magnitude_inverse * delta_y *
-        position_trajectory->speed_profile[last_element_index];
+    if (delta_x != 0)
+    {
```

Comment:
Does it matter if `delta_x` is `0` here? seems like `x_velocity_component` would just continue to be `0` either way? Similarly for `delta_y`.

Reply:
This was done so that the default values where retained if delta == 0. Otherwise the return value would be NAN

I've modified this part to be more clear.